src/theory/arith/arith_preprocess.cpp

```diff
@@ -0,0 +1,67 @@
+/*********************                                                        */
+/*! \file arith_preprocess.cpp
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Andrew Reynolds
+ ** This file is part of the CVC4 project.
+ ** Copyright (c) 2009-2020 by the authors listed in the file AUTHORS
+ ** in the top-level source directory and their institutional affiliations.
+ ** All rights reserved.  See the file COPYING in the top-level source
+ ** directory for licensing information.\endverbatim
+ **
+ ** \brief Arithmetic preprocess
+ **/
+
+#include "theory/arith/arith_preprocess.h"
+
+namespace CVC4 {
+namespace theory {
+namespace arith {
+
+ArithPreprocess::ArithPreprocess(ArithState& state,
+                                 InferenceManager& im,
+                                 ProofNodeManager* pnm,
+                                 const LogicInfo& info)
+    : d_im(im), d_opElim(pnm, info), d_reduced(state.getUserContext())
+{
+}
+TrustNode ArithPreprocess::eliminate(TNode n) { return d_opElim.eliminate(n); }
+bool ArithPreprocess::reduceAssertion(TNode atom)
+{
+  context::CDHashMap<Node, bool, NodeHashFunction>::const_iterator it =
+      d_reduced.find(atom);
+  if (it != d_reduced.end())
+  {
+    // already computed
+    return (*it).second;
+  }
+  TrustNode tn = eliminate(atom);
+  if (tn.isNull())
+  {
+    // did not reduce
+    d_reduced[atom] = false;
+    return false;
+  }
+  // tn is of kind REWRITE, turn this into a LEMMA here
```

Comment:
If it can only be `REWRITE`, we might want to assert this here. Why can't it be a conflict?

Reply:
It cant be a conflict because the lemma is of the form (= atom REDUCE(atom)) where atom does not simplify to a constant.

I'll add the assertion.